sklearn/utils/estimator_checks.py

```diff
@@ -290,6 +298,9 @@ def check_pipeline_consistency(name, Estimator):
                       random_state=0, n_features=2, cluster_std=0.1)
     X -= X.min()
     y = multioutput_estimator_convert_y_2d(name, y)
+    if name == "SpectralEmbedding":
+        # This data makes sure that the graph is fully connected.
+        X, y = S_X, S_y
```

Comment:
arpack can take a `v0` to seed the random vector. Unfortunately, passing a gaussian distributed `v0` is not principled. The correct way to do it would be to replicate:

https://github.com/scipy/scipy/blob/master/scipy/sparse/linalg/eigen/arpack/ARPACK/SRC/dgetv0.f

That looks complicated though. I am no fortran developer myself.